### ISSUE TITLE
getBigDecimal() throws exception if the decimal value has a scale greater than 0

### ISSUE THREAD

**Issue by BrendanStans21** [apache/pinot#12266]
If the decimal value in the column being passed into the getBigDecimal() method in the PinotResultSet class has a scale greater than 0, the method will throw an exception.  Any application that uses Pinot and needs to use this method to get a decimal value will fail if the value in the table has a scale greater than 0.

Suggested Solution:

Because we already have access to the value in this method, we can call a separate method before setting the scale to determine the actual scale of the value.  This way any decimal value can be retrieved using this method.  The solution would look something like this:
1. in getBigDecimal()
```
      String value = this.getString(columnIndex);
      int calculatedScale = getCalculatedScale(value)
      return value == null ? null : new BigDecimal(value).setScale(calculatedScale);
```
2. Inside the getCalculatedScale would look like this:
```
        int index = value.indexOf(".");
        return index == -1 ? 0 : value.length() - index - 1;
```

**Comment by walterddr** [comment]
this is actually a good point. 
in ResultSet getBigDecimal(String value, int scale) has been deprecated since Java 1.2. 
it seems like deriving the precision/scale from the value is the right way to go.

**Comment by Jackie-Jiang** [comment]
Fixed by #12326
